A29-xds-tls-security.md

```diff
@@ -0,0 +1,543 @@
+`CertificateProviderInstance`. Through this object the control plane tells the xDS client the name of
+a certificate and the "instance" name of a provider to use to obtain a certificate. This provider
+instance name is translated into a provider implementation and a configuration for that implementation
+using the client's own configuration instead of being sent by the control plane. This enables flexibility
+in heterogeneous deployments where different clients can use different implementations for the same
+provider instance name. For example, an on-prem client may get its certificate using a local certificate
+provider whereas a client running in cloud may use a cloud-vendor provided implementation to obtain
+the certificate.
+
+gRPC offers a `CertificateProvider` plugin API that can support multiple implementations. The plugin API
+is not currently public, so applications cannot currently add their own provider implementations although
+we might make this API public in the future. However gRPC currently includes a `file_watcher` provider
+implementation - descibed [below][FILE-WATCHER-LINK] - which reads certificates from the local file system.
+
+[FILE-WATCHER-LINK]: #file_watcher-certificate-provider
+
+Certificate provider plugin instances are configured via the xDS bootstrap file.  There is a new
+top-level field called `"certificate_providers"` whose value is a map (a JSON object).  The key
+in this map is the certificate_provider_instance name and the value is a JSON object
+having exactly 2 fields:
+
+* `"plugin_name"` which is the name of the plugin implementation (a string value), and
+* `"config"` which is the configuration for the plugin. The value of `"config"` is a JSON object
+whose schema is defined by that plugin.
+
+The structure in the bootstrap file is as follows:
+
+```jsonc
+"certificate_providers": {
+  "instance_name": {
+    "plugin_name": "implementation_name",
+    "config": {
+      // ...config for implementation_name plugin...
+    }
+  }
+}
+```
+
+This defines a plugin instance for the instance name `instance_name` which consists of a
+plugin implementation called `implementation_name` with the corresponding configuration
+defined as the value for the key `config`. When the xDS server tells the client to obtain
+a certificate from `instance_name`, the client will use this plugin instance.
+
+#### `file_watcher` Certificate Provider
+
+As mentioned before, we currently only have one plugin implementation called `file_watcher`.
+The configuration for this plugin has the following fields:
+- `certificate_file`: The path to the file containing the identity certificate. The file should contain
+a PEM-formatted X.509 conforming certificate.
+- `private_key_file`: The path to the file containing the private key. The file should contain a
+PEM-formatted PKCS encoded private key.
+- `ca_certificate_file`: The path to the file containing the root certificate. The file should contain
```

Comment:
s/certificate/certificates/. "Certificate chain" doesn't make sense here, as that is supposed to chain a leaf cert to a root cert.